### DISCUSSION TITLE
Use Obsidian Templater date module to display weeks passed since specific date?

### DISCUSSION THREAD

**Discussion by rphlacilo** [SilentVoid13/Templater#459] [Help] [answered]
Hello.

Is there a way to display a specific week number by subtracting it from the difference of today and a specified date in the past?

Use case: Week number in college semester, starting September 06 2021 until today December 2, 2021 (Week 13)

Thanks!

**Comment by slaughtr** [comment, marked as answer]
Using moment you can do this pretty easily

```
<%* 
var x = moment('2021-12-10').diff('2021-12-19', 'weeks') 
x = Math.abs(x)
tR += x
%>
```

Math.abs() isn't necessary if your first date is ahead of your second IE you're looking back, but in the above example you get `-1` - if you actually want to show negative weeks and positive, remove the Math.abs()

**Reply by rphlacilo** [reply]
Worked like a charm, thanks!

Also, for the tech uninitiated, if you want to get the week number from a specific date until today, use the following code:

```
<%*
var x = moment('2021-09-06').diff(moment(), 'weeks') 
x = Math.abs(x)
tR += x+1
%>
```

Use case: Weekly templates

**Reply by SolientArt** [reply]
I have been scouring the web for days trying to find a way to implement a "number of days since X" counter that writes the number into the markdown file. With a couple of tweaks this example looks like it does what I've been hoping to find! Thank you!

```
<%*
var x = moment().diff(moment(), 'days') 
x = Math.abs(x)
tR += x+1
%>
```
